lib/task/tools/installTask.class.php

```diff
@@ -0,0 +1,536 @@
+                [
+                    '',
+                    $e->getMessage(),
+                    '',
+                ],
+                'ERROR'
+            );
+
+            exit(2);
+        }
+
+        // Reload and test DB and ES configuration
+        $cacheClear = new sfCacheClearTask(
+            $this->dispatcher,
+            $this->formatter
+        );
+        $cacheClear->run();
+        Propel::configure($rootDir.'/config/config.php');
+        Propel::setDefaultDB('propel');
+        $this->configuration = ProjectConfiguration::getApplicationConfiguration(
+            'qubit',
+            'cli',
+            false
+        );
+        $this->context = sfContext::createInstance($this->configuration);
+        $this->context->databaseManager->loadConfiguration();
+        arElasticSearchPluginConfiguration::reloadConfig(
+            $this->context->getConfiguration()
+        );
+
+        try {
+            $this->context->getDatabaseConnection('propel');
+        } catch (Exception $e) {
+            $this->logBlock(
+                [
+                    '',
+                    'Database connection failure:',
+                    '',
+                    $e->getMessage(),
+                    '',
+                ],
+                'ERROR'
+            );
+
+            exit(2);
+        }
+
+        try {
+            arInstall::checkSearchConnection($searchOptions);
+        } catch (Exception $e) {
+            $this->logBlock(
+                [
+                    '',
+                    'Elasticsearch connection failure:',
+                    '',
+                    $e->getMessage(),
+                    '',
+                ],
+                'ERROR'
+            );
+
+            exit(2);
+        }
+
+        $this->logSection('install', 'Initializing database');
+
+        $insertSql = new sfPropelInsertSqlTask(
+            $this->dispatcher,
+            $this->formatter
+        );
+        $ret = $insertSql->run(
+            [],
+            ['no-confirmation' => $options['no-confirmation']]
+        );
+
+        // Stop when the insert SQL task is aborted
+        if ($ret) {
+            $this->logSection('install', 'Installation aborted');
+
+            exit(1);
+        }
+
+        arInstall::modifySql();
+
+        $this->logSection('install', 'Loading initial data');
+
+        arInstall::loadData($this->dispatcher, $this->formatter);
+
+        $this->logSection('install', 'Creating search index');
+
+        arInstall::populateSearchIndex();
+
+        $this->logSection('install', 'Adding site configuration');
+
+        foreach ($siteOptions as $name => $value) {
+            arInstall::createSetting($name, $value);
+        }
+
+        $this->logSection('install', 'Creating admin user');
+
+        addSuperuserTask::addSuperUser(
+            $adminOptions['username'],
+            $adminOptions
+        );
+
+        $this->logSection('install', 'Installation completed');
+    }
```

Comment:
`execute()` is > 300 lines in length. Would readability be improved if it were broken up?

Reply:
One of the downsides of being careful with the 80 chars line length and 4 chars indentation ... I try not to move things out unless they are "reused" somewhere else but I see your point. There is also some duplication between the install and purge task after configuration, so I'll see if we can avoid/improve that.

Reply:
Addressed in 40ee253 and 9ca9016.

I hope that makes it a bit clearer. We could go a bit further extending the installTask from purgeTask, but it will require to change one of the task option names and the user input process to normalize them.

Reply:
This looks great!